libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxHybridSpeechPlayer.kt

```diff
@@ -0,0 +1,91 @@
+package com.mapbox.navigation.ui.voice.api
+
+import android.content.Context
+import com.mapbox.navigation.ui.base.api.voice.SpeechPlayer
+import com.mapbox.navigation.ui.base.api.voice.SpeechPlayerCallback
+import com.mapbox.navigation.ui.base.model.voice.Announcement
+import com.mapbox.navigation.ui.base.model.voice.SpeechState
+import java.util.Queue
+import java.util.concurrent.ConcurrentLinkedQueue
+
+/**
+ * Hybrid implementation of [SpeechPlayer] combining [MapboxOnboardSpeechPlayer] and
+ * [MapboxOffboardSpeechPlayer] speech players.
+ * @property context Context
+ * @property accessToken String
+ * @property language String
+ */
+class MapboxHybridSpeechPlayer(
+    private val context: Context,
+    private val accessToken: String,
+    private val language: String
+) : SpeechPlayer {
+
+    private val queue: Queue<SpeechState.Play> = ConcurrentLinkedQueue()
+    private val offboardSpeechPlayer: MapboxOffboardSpeechPlayer =
+        MapboxOffboardSpeechPlayer(context, accessToken, language)
+    private val onboardSpeechPlayer: MapboxOnboardSpeechPlayer =
+        MapboxOnboardSpeechPlayer(context, language)
+    private var clientCallback: SpeechPlayerCallback? = null
+    private var localCallback: SpeechPlayerCallback = object : SpeechPlayerCallback {
+        override fun onDone(state: SpeechState.Done) {
+            val announcement = queue.poll()
+            clientCallback?.onDone(SpeechState.Done(announcement.announcement))
+            play()
+        }
+    }
+
+    /**
+     * Given [SpeechState.Play] [Announcement] the method will play the voice instruction.
+     * If a voice instruction is already playing or other announcement are already queued,
+     * the given voice instruction will be queued to play after.
+     * @param state SpeechState Play Announcement object including the announcement text
+     * and optionally a synthesized speech mp3.
+     * @param callback
+     */
+    override fun play(state: SpeechState.Play, callback: SpeechPlayerCallback) {
+        clientCallback = callback
+        queue.add(state)
+        if (queue.size == 1) {
+            play()
```

Comment:
if there are threading issues here, e.g., play needs to be called from a background thread

the play could only be triggered by the queue, and calling play only adds to the play queue

Reply:
`@UiThread`